lib/libyui/tasks.rb

```diff
@@ -13,16 +13,31 @@
 # License along with this library; if not, write to the Free Software
 # Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
 #++
+require "yaml"
+
 module Libyui
   module Tasks
     # Name of the CMake version file
     VERSION_CMAKE = "VERSION.cmake"
+    # Targets definition
+    TARGETS_FILE = File.expand_path("../../../data/targets.yml", __FILE__)
 
     # Wrapper to set up packaging tasks
     def self.configuration(&block)
       ::Packaging.configuration(&block)
     end
 
+    def self.submit_to(target, file = TARGETS_FILE)
+      targets = YAML.load_file(file)
+      config = targets[target]
+      raise "Not configuration found for #{target}" if config.nil?
+      Libyui::Tasks.configuration do |conf|
+        config.each do |meth, val|
+          conf.send("#{meth}=", val)
```

Comment:
for new code I prefer to use http://ruby-doc.org/core-2.2.3/Object.html#method-i-public_send as it is more safe, even if data is our own, it is good for robustnest of appliance.